pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_agent.py

```diff
@@ -153,15 +175,33 @@ async def streamed_response():
         )
         activities.append(self.event_stream_handler_activity)
 
+        if additional_models:
+            if isinstance(additional_models, Mapping):
+                for key, value in additional_models.items():
+                    self._register_additional_model(key, value)
+            else:
+                for value in additional_models:
+                    if not isinstance(value, str):
+                        raise UserError(
+                            'When providing `additional_models` as a sequence, all entries must be model strings like "openai:gpt-4o". '
+                            'To register Model instances, pass a mapping of names to models.'
+                        )
+                    self._register_string_model(value, value)
+
         temporal_model = TemporalModel(
-            wrapped.model,
+            wrapped_model,
             activity_name_prefix=activity_name_prefix,
             activity_config=activity_config | model_activity_config,
             deps_type=self.deps_type,
             run_context_type=self.run_context_type,
             event_stream_handler=self.event_stream_handler,
+            model_selection_var=self._model_selection_var,
```

Comment:
Instead of having a context var that's owned at this level and then passed into the `TemporalModel`, could `TemporalModel` own the context var and have a contextmanager method that we could use like `with self._temporal_model.using_model('...'):`?